src/frontend/src/storage/index.ts

```diff
@@ -269,18 +280,46 @@ const nowMillis = (): number => {
  * user's browser is compromised: the principals associated with and origin of the visited dapps
  * should in principle not be leaked.
  *
- * The digest is a sha hash of the origin, principal and some secret, i.e. a HMAC digest (with
- * non-extractable HMAC key). The origin is hashed in to further reduce the already unlikely
- * possibility of collisions.
+ * The digest is a sha hash of the principal and some secret, i.e. a HMAC digest (with
+ * non-extractable HMAC key).
  */
 const computePrincipalDigest = async ({
-  origin: origin_,
   principal: principalObj,
   hasher,
 }: {
-  origin: string;
   principal: Principal;
   hasher: CryptoKey;
+}): Promise<string> => {
+  // Encode each element & size
+  const enc = new TextEncoder();
+  const principal_ = principalObj.toText();
+
+  const principal = enc.encode(principal_);
+  const principalLen = Uint8Array.from([principal_.length]);
+
+  // Create a buffer with all four elements
```

Comment:
@frederikrothenberger `with two elements`

Reply:
Thanks, addressed in #2581.